tests/cli-misc.rs

```diff
@@ -905,3 +905,14 @@ fn which() {
         );
     });
 }
+
+#[test]
+fn delete_downloads_dir_contents_by_update() {
```

Comment:
This test case should explicitly *NOT* clean up the downloads directory because there might be partials for updating a different toolchain which it would be bad to lose access to.